File: app/src/main/java/wangdaye/com/geometricweather/db/generators/MinutelyEntityGenerator.java
package wangdaye.com.geometricweather.db.generators;

import java.util.ArrayList;
import java.util.List;

import wangdaye.com.geometricweather.basic.models.options.provider.WeatherSource;
import wangdaye.com.geometricweather.basic.models.weather.Minutely;
import wangdaye.com.geometricweather.db.entities.MinutelyEntity;
import wangdaye.com.geometricweather.db.converters.WeatherSourceConverter;

public class MinutelyEntityGenerator {

    public static MinutelyEntity generate(String cityId, WeatherSource source, Minutely minutely) {
        MinutelyEntity entity = new MinutelyEntity();

        entity.cityId = cityId;
        entity.weatherSource = new WeatherSourceConverter().convertToDatabaseValue(source);
        
        entity.date = minutely.getDate();
        entity.time = minutely.getTime();
        entity.daylight = minutely.isDaylight();

        entity.weatherCode = minutely.getWeatherCode();
        entity.weatherText = minutely.getWeatherText();

        entity.minuteInterval = minutely.getMinuteInterval();
        entity.dbz = minutely.getDbz();
        entity.cloudCover = minutely.getCloudCover();

        return entity;
    }

    public static List<MinutelyEntity> generate(String cityId, WeatherSource source,
                                                List<Minutely> minutelyList) {
        List<MinutelyEntity> entityList = new ArrayList<>(minutelyList.size());
        for (Minutely minutely : minutelyList) {
            entityList.add(generate(cityId, source, minutely));
        }
        return entityList;
    }

    public static Minutely generate(MinutelyEntity entity) {
        return new Minutely(
                entity.date, entity.time, entity.daylight,
                entity.weatherText, entity.weatherCode,
                entity.minuteInterval, entity.dbz, entity.cloudCover
        );
    }

    public static List<Minutely> generate(List<MinutelyEntity> entityList) {
        List<Minutely> dailyList = new ArrayList<>(entityList.size());
        for (MinutelyEntity entity : entityList) {
            dailyList.add(generate(entity));
        }
        return dailyList;
    }
}

File: app/src/main/java/org/breezyweather/db/generators/MinutelyEntityGenerator.kt
/**
 * This file is part of Breezy Weather.
 *
 * Breezy Weather is free software: you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as published by the
 * Free Software Foundation, version 3 of the License.
 *
 * Breezy Weather is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
 * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
 * License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Breezy Weather. If not, see <https://www.gnu.org/licenses/>.
 */

package org.breezyweather.db.generators

import org.breezyweather.common.basic.models.weather.Minutely
import org.breezyweather.db.entities.MinutelyEntity

object MinutelyEntityGenerator {
    fun generate(formattedId: String, minutely: Minutely): MinutelyEntity {
        return MinutelyEntity(
            formattedId = formattedId,
            date = minutely.date,
            minuteInterval = minutely.minuteInterval,
            dbz = minutely.dbz
        )
    }

    fun generate(formattedId: String, minutelyList: List<Minutely>): List<MinutelyEntity> {
        val entityList: MutableList<MinutelyEntity> = ArrayList(minutelyList.size)
        for (minutely in minutelyList) {
            entityList.add(generate(formattedId, minutely))
        }
        return entityList
    }

    fun generate(entity: MinutelyEntity): Minutely {
        return Minutely(
            entity.date,
            entity.minuteInterval,
            entity.dbz
        )
    }

    fun generate(entityList: List<MinutelyEntity>): List<Minutely> {
        val dailyList: MutableList<Minutely> = ArrayList(entityList.size)
        for (entity in entityList) {
            dailyList.add(generate(entity))
        }
        return dailyList
    }

}
